src/controllers/users.js

```diff
@@ -188,6 +190,61 @@ class UserController {
       res
     );
   }
+
+  /**
+   * @name updateUserInfo
+   * @description Updates user profile to complete registration
+   * @param {object} req The request object
+   * @param {object} res The response object
+   * @returns {object} The API response
+   */
+  async updateUserInfo(req, res) {
+    try {
+      const userData = { ...req.body };
+      const { user } = res.locals;
+
+      userData.password = hash.generateSync(userData.password);
+      const data = await UserServices.updateUserInfoByEmail({ ...userData }, user.email);
+
+      if (data === 'own_manage') {
+        return Responses.handleError(409, 'You cannot manage your self', res);
+      } else {
+        return Responses.handleSuccess(201, 'Updated successfully', res, data);
+      }
+    } catch (error) {
+      return Responses.handleError(500, error.message, res);
+    }
+  }
+
+  /**
+   * @name getUserProfile
+   * @description gets user profile to complete registration
+   * @param {object} req The request object
+   * @param {object} res The response object
+   * @returns {object} The API response
+   */
+  async getUserProfile(req, res) {
+    let { userId } = req.params;
+
+    userId = Number(userId);
+
+    if (typeof userId === 'number' && userId > 0 && userId >>> 0 === userId) {
+      const { userData, lineManagerData } = await UserServices.getUserById(userId);
+
+      if (userData !== null) {
+        const user = userData.dataValues;
+
+        user.lineManager = lineManagerData ? `${lineManagerData.dataValues.firstName} ${lineManagerData.dataValues.lastName}` : 'none';
+
+        const { password, lineManagerId, ...data } = user;
+
+        return Responses.handleSuccess(200, 'Profile retrieved!', res, data);
+      } else {
```

Comment:
no need for this `else` statement.

Reply:
You're right, let me change it that way.